lib/project_types/theme/models/specification_handlers/theme.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Theme
+  module Models
+    module SpecificationHandlers
+      class Theme
+        REQUIRED_FOLDERS = %w(config layout sections templates).map { |folder| "#{folder}/" }
+
+        def initialize(root)
```

Comment:
An IO operation as part of instantiating an object might be an unintended side-effect for someone using this class. What about adjusting `initialize` to be initialized with the folders and have a static method for initializing it by doing the IO operation in the system

```rb
class Theme
  def constructor(folders)
  end
  def self.new_from_theme_directory(directory)
  end
end
```

Reply:
I will keep this in mind for future PRs, but with my latest code changes, IO operations have moved from the constructor :)